src/main/java/com/abitmorecode/songrest/Services/SongService.java

```diff
@@ -0,0 +1,220 @@
+package com.abitmorecode.songrest.Services;
+
+import com.abitmorecode.songrest.Controller.SongController;
+import com.abitmorecode.songrest.Models.Song;
+import com.abitmorecode.songrest.SongControllerException.NoIdAvailableException;
+import com.abitmorecode.songrest.SongControllerException.SameSongAlreadyExistException;
+import com.abitmorecode.songrest.SongControllerException.SongDoesntExistException;
+import com.abitmorecode.songrest.SongControllerException.SongIdAlreadyExistException;
+import com.google.gson.Gson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Song Controller (Service)
+ */
+
+@Service
+public class SongService implements SongsManager {
+
+	private static final Logger log = LoggerFactory.getLogger(SongController.class);
+
+	private static final List<Song> songs = new ArrayList<>();
+
+	private static final Gson gson = new Gson();
+
+	/**
+	 * default constructor
+	 */
+	public SongService() {
+		String filepath = "exercises/1/songs.json";
```

Comment:
JSON sollte sich in Resource Ordner befindenund daraus geladen werden mit z.B. `  URL resource = getClass().getClassLoader().getResource("file.txt")`